core/src/test/java/ai/timefold/solver/core/impl/score/stream/common/AbstractFactChangePropagationTest.java

```diff
@@ -1,81 +0,0 @@
-package ai.timefold.solver.core.impl.score.stream.common;
-
-import static org.assertj.core.api.Assertions.assertThat;
-
-import java.util.List;
-
-import ai.timefold.solver.core.api.domain.entity.PlanningEntity;
-import ai.timefold.solver.core.api.score.buildin.simple.SimpleScore;
-import ai.timefold.solver.core.api.score.stream.Constraint;
-import ai.timefold.solver.core.api.score.stream.ConstraintFactory;
-import ai.timefold.solver.core.api.score.stream.ConstraintProvider;
-import ai.timefold.solver.core.api.solver.Solver;
-import ai.timefold.solver.core.api.solver.SolverFactory;
-import ai.timefold.solver.core.config.constructionheuristic.ConstructionHeuristicPhaseConfig;
-import ai.timefold.solver.core.config.solver.SolverConfig;
-import ai.timefold.solver.core.impl.testdata.domain.chained.shadow.TestdataShadowingChainedAnchor;
-import ai.timefold.solver.core.impl.testdata.domain.chained.shadow.TestdataShadowingChainedEntity;
-import ai.timefold.solver.core.impl.testdata.domain.chained.shadow.TestdataShadowingChainedSolution;
-
-import org.jspecify.annotations.NonNull;
-import org.junit.jupiter.api.Test;
-
-public abstract class AbstractFactChangePropagationTest {
-
-    private static final String VALUE_CODE = "v1";
-
-    private final Solver<TestdataShadowingChainedSolution> solver;
-
-    protected AbstractFactChangePropagationTest() {
-        this.solver = buildSolver();
-    }
-
-    /**
-     * Tests if Drools try to modify a {@link PlanningEntity} before its shadow variables are updated.
```

Comment:
Test removed for two reasons:

- It tested a situation that happened in Drools, and we haven't supported Drools in years.
- The situation being tested is no longer possible with this PR, because the solver will terminate before it could get here.